### DISCUSSION TITLE
Multiple queries regarding Sunbird

### DISCUSSION THREAD

**Discussion by vishaldeltech** [Sunbird-Ed/Community#258] [General Discussion]
Hi @parthlawate, as suggested, here are the queries discussed in our meeting

### Sunbird Multi-tenant
- Multitenancy in sunbird is mentioned at 25:18 in this video: [Understanding Content Architecture of Sunbird](https://www.youtube.com/watch?v=WxZXaTnj2D0&ab_channel=Sunbird) but the documentation and community discussions suggest otherwise - https://github.com/project-sunbird/sunbird-community/discussions/591]]. The GitHub link talks about a solution but additional inputs would be helpful. 

### Sunbird Lern
- Tenants have the ability to override configurations. Can they override to expand the limits set via system admins?
- What can tenants override? can they add their own frameworks that are not in the core?
- Merit certificate capabilities - can a combination of conditions be written? For example, a learner needs 80% in 5 specific assessments, and >75% attendance to earn a certificate
- Can we have batches independent of mentors? We have use cases where 1 batch can have multiple changing mentors.
- Can we have multiple concurrent active batches for a course? Already raised [here](https://github.com/orgs/Sunbird-Ed/discussions/155) but more info would be helpful
- Can we have indefinite batches (no end date)?
- Does telemetry capture time spent on a specific question in question sets?
- Can we disable learner self-registration to a batch, and instead allow tenant staff to add/remove learners from a batch. This [link](https://github.com/orgs/Sunbird-Ed/discussions/155) talk about invite only courses. More info would be helpful
- How does mixed taxonomy work? In one of the videos on youtube, the experts said that it is possible to add collections of different taxonomies to another. Example: If I add 2 collections of different taxonomies to a third collection, how will the hierarchy/nomenclature behave?

### Other questions that were discussed in the meeting (for reference), with the responses in _italics_
- Can a batch have multiple courses? - _Not available in the UI_
- Optional content in a course (this is not mandatory to complete the course) - _Not available_
- Pre-requisites for a course are trackable but not mandatory for course completion. Is this feature available? - _Not available_
- Can editors/publishers access any course in the tenant? Is it possible to limit them to being able to create/edit/publish courses for specific domains / tag / business unit? - _Not available_
- How are frameworks added/updated? Videos described that it is done in a no-code way - _This is done via API. No UI is available right now_
- PGP, Masters - these are groups of courses that are sold as a single entity. Is this feature available out of the box? - _Not available_
- Is it possible to give different progress weights to subparts of a collection or at the content level? Example use case: courses in a PGP program have credits that a learner needs to earn - _Not available_
- Can we make custom changes in the course for a specific batch? For example, add class recordings that are accessible only by that class. - _Not available_
- Custom sections and pedagogy in a course but only for a specific batch/s - _Not available_
- Lock unlock / due dates for cohorts (relative and absolute dates, completion of specific content, etc) - _Not available_
- Do learners see all published courses on sunbird or only those they have access to via the cohort? - _They see all the courses_
- Can only the cohort manager add/remove learners from a cohort? - _Feature not available_
- User location - State and district are mandatory. Can it be made non-mandatory via a config file? - _Yes they can be made non-mandatory_
- Question set player - time limit is available for questions, but is it available for sections? - _Not available_
- n-level depth of sections available in question sets? _Yes_
- Ability to merge or split batches? If this is available out of the box, how is the progress managed? - _Not available_
- How does batch management permission hierarchy work? Example: The branch manager of a coaching institute can only manage batches only for their own branch

**Comment by Krishnaj20** [comment]
thanks @vishaldeltech for initiating this thread.
Tagging @vinukumar-vs @rhwarrier @parthlawate @sharathkashyap @vrayulu @reshmi-nair

**Reply by parthlawate** [reply]
@Krishnaj20 do you have access to split this into multiple threads (assuming github allows that) so it will be easier to give replies ?

**Comment by Krishnaj20** [comment]
@vishaldeltech & team - It was great connecting with you and the team. As discussed in today's meeting please post the open question on custom roles and permissions. We will close it here.
@vrayulu
